nautobot_golden_config/templates/nautobot_golden_config/configplan_create.html

```diff
@@ -65,9 +65,40 @@
   delete form_data.q;
   return {"data": form_data};
 }
-var jobClass = "plugins/nautobot_golden_config.jobs/GenerateConfigPlans"
-var redirectUrlTemplate = "/plugins/golden-config/config-plan/?job_result_id={jobData.result.id}"
-document.getElementById("startJob").onclick = function() {startJob(jobClass, formatJobData($("form").serializeArray()), redirectUrlTemplate)};
+
+function isValidURL(url) {
+  // Regular expression to validate URLs
+  var urlPattern = /^(https?:\/\/)?([\w-]+(\.[\w-]+)+|localhost|(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}))(:\d+)?(\/\S*)?$/;
+  return urlPattern.test(url);
+}
+
+function openModalAndStartJob() {
+  var changeControlUrl = changeControlUrlInput.value;
+  if (!isValidURL(changeControlUrl)) {
```

Comment:
```suggestion
  if (changeControlUrl && !isValidURL(changeControlUrl)) {
```
url should be able to be blank also. so blank is valid, otherwise valid url is required.